pulsar-client-api/src/main/java/org/apache/pulsar/client/api/DeadLetterPolicy.java

```diff
@@ -49,4 +49,15 @@
      */
     private String deadLetterTopic;
 
+    /**
+     * Name of the initial subscription name of the dead letter topic.
+     * The default value is the subscription name of the consumer.
+     */
+    private String initSubscriptionName;
```

Comment:
@RobertIndie Since we're adding new public API, we should also write this proposal down in a PIP.